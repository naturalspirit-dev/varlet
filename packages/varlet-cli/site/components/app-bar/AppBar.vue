<template>
  <div
    class="var-site-app-bar"
    :class="{ 'var-site-elevation--3': elevation }"
    :style="{
      background: color,
      color: textColor,
    }"
  >
    <div class="var-site-app-bar__left" v-if="$slots.left">
      <slot name="left" />
    </div>

    <div
      class="var-site-app-bar__title"
      :class="`var-site-app-bar__title-${titlePosition}`"
      :style="{
        paddingLeft: $slots.left && titlePosition !== 'center' ? '30px' : '10px',
        paddingRight: $slots.right && titlePosition !== 'center' ? '30px' : '10px',
      }"
    >
      <slot>{{ title }}</slot>
    </div>

    <div class="var-site-app-bar__right" v-if="$slots.right">
      <slot name="right" />
    </div>
  </div>
</template>

<script lang="ts">
import { defineComponent } from 'vue'
import { props } from './props'

export default defineComponent({
  name: 'VarSiteAppBar',
  props,
})
</script>

<style lang="less">
@import '../styles/common';
@import '../styles/elevation';
@import './appBar';
</style>
