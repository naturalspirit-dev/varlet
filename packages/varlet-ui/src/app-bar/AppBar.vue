<template>
  <div
    class="var-app-bar"
    :class="{ 'var-elevation--3': elevation }"
    :style="{
      background: color,
      color: textColor,
    }"
  >
    <div class="var-app-bar__left" v-if="$slots.left">
      <slot name="left"></slot>
    </div>

    <div
      class="var-app-bar__title"
      :class="`var-app-bar__title-${titlePosition}`"
      :style="{
        paddingLeft: $slots.left && titlePosition !== 'center' ? '30px' : '10px',
        paddingRight: $slots.right && titlePosition !== 'center' ? '30px' : '10px',
      }"
    >
      <slot>{{ title }}</slot>
    </div>

    <div class="var-app-bar__right" v-if="$slots.right">
      <slot name="right"></slot>
    </div>
  </div>
</template>

<script lang="ts">
import { defineComponent } from 'vue'
import { props } from './props'
import Icon from '../icon'

export default defineComponent({
  name: 'VarAppBar',
  components: {
    [Icon.name]: Icon,
  },
  props,
})
</script>

<style lang="less">
@import '../icon/icon';
@import './appBar';
</style>
