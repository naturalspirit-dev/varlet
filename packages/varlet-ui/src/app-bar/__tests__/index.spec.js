import example from '../example'
import AppBar from '..'
import VarAppBar from '../AppBar'
import { mount } from '@vue/test-utils'
import { createApp } from 'vue'

test('test appBar example', () => {
  const wrapper = mount(example)

  expect(wrapper.html()).toMatchSnapshot()
})

test('test appBar plugin', () => {
  const app = createApp({}).use(AppBar)
  expect(app.component(AppBar.name)).toBeTruthy()
})

test('test appBar props', () => {
  const wrapper = mount(VarAppBar, {
    props: {
      color: '#a3b9db',
      textColor: '#000',
      title: 'test title',
      titlePosition: 'center',
      elevation: 'false',
    },
    slots: {
      left: () => 'leftSlot',
      right: () => 'rightSlot',
    }
  })


  expect(wrapper.html()).toMatchSnapshot()
})


