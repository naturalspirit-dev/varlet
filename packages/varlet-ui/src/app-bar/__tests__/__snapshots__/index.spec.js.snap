// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test app bar example 1`] = `
"<div class=\\"app-type\\">基础导航栏</div>
<div class=\\"var-app-bar var-elevation--3\\" style=\\"background: rgb(41, 121, 255); color: rgb(255, 255, 255);\\">
  <!--v-if-->
  <div class=\\"var-app-bar__title var-app-bar__title-left\\" style=\\"padding-left: 10px; padding-right: 10px;\\">标题</div>
  <!--v-if-->
</div>
<div class=\\"app-type\\">自定义样式</div>
<div class=\\"var-app-bar var-elevation--3\\" style=\\"background: rgb(255, 159, 0); color: rgb(255, 255, 255);\\">
  <!--v-if-->
  <div class=\\"var-app-bar__title var-app-bar__title-center\\" style=\\"padding-left: 10px; padding-right: 10px;\\">标题</div>
  <!--v-if-->
</div>
<div class=\\"app-type\\">添加标题处插槽</div>
<div class=\\"var-app-bar var-elevation--3\\" style=\\"background: rgb(41, 121, 255); color: rgb(255, 255, 255);\\">
  <!--v-if-->
  <div class=\\"var-app-bar__title var-app-bar__title-left\\" style=\\"padding-left: 10px; padding-right: 10px;\\">从插槽处添加标题</div>
  <!--v-if-->
</div>
<div class=\\"app-type\\">添加左侧插槽</div>
<div class=\\"var-app-bar var-elevation--3\\" style=\\"background: rgb(41, 121, 255); color: rgb(255, 255, 255);\\">
  <div class=\\"var-app-bar__left\\"><button class=\\"var-button var--box var-button--normal var--inline-flex var-button--text-default var-button--text var-button--round\\" style=\\"color: rgb(255, 255, 255); background: transparent;\\">
      <!--v-if-->
      <div class=\\"var-button__content\\"><i class=\\"var-icon var-icon--set var-icon-chevron-left\\" style=\\"transition: all 0ms; font-size: 24px;\\"></i></div>
    </button></div>
  <div class=\\"var-app-bar__title var-app-bar__title-center\\" style=\\"padding-left: 10px; padding-right: 10px;\\">标题</div>
  <!--v-if-->
</div>
<div class=\\"app-type\\">添加右侧插槽</div>
<div class=\\"var-app-bar var-elevation--3\\" style=\\"background: rgb(41, 121, 255); color: rgb(255, 255, 255);\\">
  <!--v-if-->
  <div class=\\"var-app-bar__title var-app-bar__title-left\\" style=\\"padding-left: 10px; padding-right: 30px;\\">标题</div>
  <div class=\\"var-app-bar__right\\"><button class=\\"var-button var--box var-button--normal var--inline-flex var-button--text-default var-button--text var-button--round\\" style=\\"color: rgb(255, 255, 255); background: transparent;\\">
      <!--v-if-->
      <div class=\\"var-button__content\\"><i class=\\"var-icon var-icon--set var-icon-magnify\\" style=\\"transition: all 0ms; font-size: 24px;\\"></i></div>
    </button></div>
</div>
<div class=\\"app-type\\">添加左右插槽</div>
<div class=\\"var-app-bar var-elevation--3\\" style=\\"background: rgb(41, 121, 255); color: rgb(255, 255, 255);\\">
  <div class=\\"var-app-bar__left\\"><button class=\\"var-button var--box var-button--normal var--inline-flex var-button--text-default var-button--text var-button--round\\" style=\\"color: rgb(255, 255, 255); background: transparent;\\">
      <!--v-if-->
      <div class=\\"var-button__content\\"><i class=\\"var-icon var-icon--set var-icon-chevron-left\\" style=\\"transition: all 0ms; font-size: 24px;\\"></i></div>
    </button></div>
  <div class=\\"var-app-bar__title var-app-bar__title-left\\" style=\\"padding-left: 30px; padding-right: 30px;\\">标题</div>
  <div class=\\"var-app-bar__right\\">
    <div class=\\"var-menu\\"><button class=\\"var-button var--box var-button--normal var--inline-flex var-button--text-default var-button--text var-button--round\\" style=\\"color: rgb(255, 255, 255); background: transparent;\\">
        <!--v-if-->
        <div class=\\"var-button__content\\"><i class=\\"var-icon var-icon--set var-icon-menu\\" style=\\"transition: all 0ms; font-size: 24px;\\"></i></div>
      </button>
      <!--teleport start-->
      <!--teleport end-->
    </div>
  </div>
</div>
<div class=\\"space\\"></div>"
`;

exports[`test app bar props 1`] = `
"<div class=\\"var-app-bar\\" style=\\"background: rgb(163, 185, 219); color: rgb(0, 0, 0);\\">
  <div class=\\"var-app-bar__left\\">leftSlot</div>
  <div class=\\"var-app-bar__title var-app-bar__title-center\\" style=\\"padding-left: 10px; padding-right: 10px;\\">test title</div>
  <div class=\\"var-app-bar__right\\">rightSlot</div>
</div>"
`;
